application/frontend/controllers/ResetController.php

```diff
@@ -81,9 +127,77 @@ public function actionResend($uid)
             throw new NotFoundHttpException();
         }
 
+        /*
+         * Resend verification
+         */
         $reset->send();
 
-        \Yii::$app->response->statusCode = 204;
-        return;
+        return $reset;
+    }
+
+    /**
+     * Validate reset code. Logs user in if successful
+     * @param string $uid
+     * @return Reset
+     * @throws BadRequestHttpException
+     * @throws NotFoundHttpException
+     * @throws ServerErrorHttpException
+     * @throws \Exception
+     */
+    public function actionValidate($uid)
+    {
+        /** @var Reset $reset */
+        $reset = Reset::findOne(['uid' => $uid]);
+        if ($reset === null) {
+            throw new NotFoundHttpException();
+        }
+
+        $code = \Yii::$app->request->getBodyParam('code', null);
+        if ($code === null) {
+            throw new BadRequestHttpException('Code is required', 1462989866);
+        }
+
+        $log = [
+            'action' => 'Validate reset',
+            'reset_id' => $reset->id,
+        ];
+
+        $isValid = $reset->isUserProvidedCodeCorrect($code);
+        if ($isValid === true) {
```

Comment:
I wanted to have an explicit check and not a loose check. Since that determines whether what they provided is correct or not, I want to be sure it is === true and not something considered truthy. I could do all that in the if call without variable assignment, but I thought the assignment made it more readable.